src/game/common/system/upgrade.cpp

```diff
@@ -33,19 +33,23 @@ UpgradeCenter::~UpgradeCenter()
 
 void UpgradeCenter::Init()
 {
-#ifdef GAME_DLL
-    Call_Method<void, UpgradeCenter>(PICK_ADDRESS(0x004AEF70, 0x00728CDD), this);
-#endif
+    New_Upgrade("")->Friend_Make_Veterancy_Upgrade(VETERANCY_VETERAN);
+    New_Upgrade("")->Friend_Make_Veterancy_Upgrade(VETERANCY_ELITE);
+    New_Upgrade("")->Friend_Make_Veterancy_Upgrade(VETERANCY_HEROIC);
 }
 
 void UpgradeCenter::Reset()
 {
-#ifdef GAME_DLL
-    Call_Method<void, UpgradeCenter>(PICK_ADDRESS(0x004AF190, 0x00728DBD), this);
-#endif
+    if (g_theMappedImageCollection != nullptr && !m_buttonImagesCached) {
+        for (UpgradeTemplate *upgrade = m_upgradeList; upgrade != nullptr; upgrade = upgrade->Friend_Get_Next()) {
+            upgrade->Cache_Button_Image();
+        }
+
+        m_buttonImagesCached = true;
```

Comment:
This is surprising. Should the caching not happen in the Init?

Reply:
This is correct to the original.